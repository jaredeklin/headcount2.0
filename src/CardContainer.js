import React from 'react';
import Card from './Card';
import './styles/cardContainer.css';
import PropTypes from 'prop-types'

export const CardContainer = ({data, selectCard}) => {

  const cardInfo = data.map(district => {
    return (
      <Card location={district.location} 
            data={district.data} 
            key={district.location} 
            selectCard={selectCard}
      />)
  })
  
  return (
    <div className="card-container">
      {cardInfo}
    </div>
  )
}

CardContainer.propTypes = {
  data: PropTypes.array.isRequired,
  selectCard: PropTypes.func
}