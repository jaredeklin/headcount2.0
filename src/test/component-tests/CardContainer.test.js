import React from 'react';
import { shallow } from 'enzyme';
import { CardContainer } from '../../CardContainer';

describe('CardContainer', () => {
  
  it('should match the snapshot', () => {
    let mockData = [{
      '2004': 0.24,
      '2005': 0.278
    }];
    let wrapper = 
      shallow(<CardContainer 
        stats={mockData} 
        selectedCards={mockData}
      />);

    expect(wrapper).toMatchSnapshot();
  });

});