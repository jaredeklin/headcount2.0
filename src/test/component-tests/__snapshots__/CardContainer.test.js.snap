// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardContainer should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    selectedCards={
        Array [
            Object {
              "2004": 0.24,
              "2005": 0.278,
            },
          ]
    }
    stats={
        Array [
            Object {
              "2004": 0.24,
              "2005": 0.278,
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Card
          className="card clicked"
          location={undefined}
          selectCard={undefined}
          statType={undefined}
          stats={undefined}
/>,
      ],
      "className": "card-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "class",
        "props": Object {
          "className": "card clicked",
          "location": undefined,
          "selectCard": undefined,
          "statType": undefined,
          "stats": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Card
            className="card clicked"
            location={undefined}
            selectCard={undefined}
            statType={undefined}
            stats={undefined}
/>,
        ],
        "className": "card-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "class",
          "props": Object {
            "className": "card clicked",
            "location": undefined,
            "selectCard": undefined,
            "statType": undefined,
            "stats": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
